### PR TITLE
Expert progression

### PR DESCRIPTION
**Pull request by MuteTiefling**
## Getting Started
Primitive Ore Processing

I'm avoiding getting too in depth with the scripts for this for now. That'll be a job for when EE comes into play.
Ore cannot be smelted directly and only "Simple" ores will be able to be processed for now. This includes:

| Ore    | Primary Output | Secondary Output |
|--------|----------------|------------------|
| Copper | Copper         | Nickel           |
| Silver | Silver         | Lead             |
| Gold   | Gold           | Copper           |
| Tin    | Tin            | Quartz           |

Make a Pestle and Mortar. This will enable grinding ore to start. 
image

Make a Furnace, and then a Blast Furnace
image
image

Working with Simple Metals
Crush copper
image

Smelt down to Clump
image

Crush again to dust
image

Finally Blast to Ingot
image

Alternative recipes planned for later in progression. I'm not currently happy with the double crush here. I was hoping to have Tinkers by now and the final step of this would have been to melt/cast rather than crush again and cook. Ore processing is still very much a work in progress.

## Getting Witchy
With some simple metallurgy, we can start getting down to business. The player will start with leather armor, a bow, and a quiver full of arrows. Their first task will be to kill the Naga.
image

Extra scales have been added to the loot table as well as some extras that are not otherwise accessible in Twilight Forest.
image

Hexerei forms the basis for a lot of crafts, so we'll make a cauldron
image
And right click it with a Naga scale to convert it to a Mixing Cauldron
image

## A Touch of Druidic Magic
Getting into Nature's Aura early will just need some string and a Firefly, easily gathered from the starting area.
image

Ritual of the Forest Wooden Stands are crafted by right clicking any log with the Gold Leaf
image

Ritual of the Forest enables access to Archwood
image
And other saplings
image
image

With Archwood we can get an Imbuement Chamber going.
image
Giving access to Infused Stone
image

## Gathering the Elements
Elemental Essences will require some exploration through Twilight Forest.
Currently each of these needs an item from Thermal/Vanilla elementals. These elementals spawn in the first wave of a Gateway Rift that occurs 25% of the time when breaking a spawner. There's a very good chance another method of summoning them is required, but that's easily done with a ritual. Can summon in the existing Elemental Gateways to make them accessible. 

As a note, I did try adding them as naturally spawning in Twilight Forest, but even with a very low weight they ended up swarming the whole place. It was... not pretty. 

Exploring the starting areas
image
To the Fire Swamp to defeat the hydra
image
A trip to the Troll Caves
image
Defeating the Yeti
image

Other essences have also been tweaked to be accessible
Manipulation and Conjuration essences are used a lot, and Abjuration Essences are used in place of Nether Wart for potions.
image
image
image

## Drawing on the Occult
Rituals will be used heavily throughout the pack, and several custom pentacles are already in place with more planned. These starting ones avoid the 'otherworld spirits' for now, as those will be a major theme in later progression.

Getting some Chalk going for rituals
image
image

Potions will be very important for both crafting and for generating Aura and Source. Potions of Ancient Knowledge have a few paths, but will most likely come from Ancient Fruit (Quark's Ancient Sapling)

Ancient Knowledge Potions
image
image

Magebloom Seed
This'll be important for making Magebloom Fiber
image

Source come from Potions and the Alchemical Sourcelink. 
image

Aura will come from the Lingering Absorber to start, requiring a visit to the UrGhast Tower for Netherrack
image
Everything for the multiblock is craftable from resources easily found there. 

## Calling on Higher Powers
Make an Offering Table
image

Make some Ironwood for tools and equipment, as well as other crafting. Ironwood will be very important early on and represents a major equipment upgrade. 
image
image

With Ironwood, we can gain access to the Infusion Altar and start working with Aura directly. 
image

## A Side of Technology
Andesite Alloy is  removed entirely and is replaced by Compressed Stone in recipes.
Dried Kelp also removed from Create recipes, replaced with leather. Kelp is not available in Twilight Forest as there are no oceans. 
image
image
Ersatz Leather from IE allows for a nice plant based option still. 
image
A great deal of recipes still need tweaking in Create as many things still use Iron, which is unavailable. 

Compressed Stone is made by exploding Andesite. 50% loss rate.
image

Using magic and Infused Rock, the conversion rate goes up to 100% 
image

## Under Pressure
Tier 1 Pneumaticcraft will feature into the early game, prior to any real power generation. Indeed, it will be critical in opening up AE2 and basic power generation.

Making Pressure:
The basic Air Compressor will be the starting point. With a little magic from Lychee it will consume a small amount of Aura randomly while operating. The lore being that heating the Air Essence leads it to draw in Aura and produce compressed air. 

image
image

The next step from here will be straight to the FE compressor, not long after we start making power. That'll need Iron, however, which is a major goal for the Blue Skies dims.

Most basic parts should be accessible at this point, with Compressed Iron replaced by Compressed Stone in these early stages.

Pressure Chamber
image
Pressure Gauge
image

## Getting Organized
With Basic PNC and some magic, we can start getting an AE2 system going.
First logistics cores:
image
Crafting Terminals
image
image

Quartz Fiber
image
Fluix ME Glass Cable unchanged for now. Likely forever, want it to remain simple
image
Energy Acceptor
image
Crystal Growth Accelerator
image
ME Interface
image
ME Storage Bus
image

## Making Power
The Energetic Aura Forge will be required to start converting Aura to FE. This means climbing the UrGhast tower for some materials
image

### CODE REVIEW

**Comment by MuteTiefling** [comment]
Non progression changes
Hoppers made from Tin Plates rather than Iron as Iron will come much later
image


Engineer's Hammer for early plates made of lead: 
image

**Comment by NielsPilgaard** [comment]
I'm currently on vacation, would it be a pain if I didn't review this before Friday? 
Looking forward to reading through this though 😃

**Comment by MuteTiefling** [comment]
No trouble at all. Enjoy!

I'll take the time to make sure the read through on github is a little more organized.

**Comment by NielsPilgaard** [comment]
Thanks ❤

**Review by NielsPilgaard — APPROVED** [review]

File: kubejs/server_scripts/expert/player_events/starting_items.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    if (!event.player.stages.has('starting_items')) {
+        event.player.stages.add('starting_items');
+
+        let starting_items = [
+            Item.of(
+                'minecraft:bow',
+                '{Damage:0,"quark:RuneAttached":1b,"quark:RuneColor":{Count:1b,id:"quark:blank_rune"}}'
+            ).enchant('minecraft:unbreaking', 1),
```
[the remark this answers is not in the document]
Comment (MuteTiefling) [reply kubejs/server_scripts/expert/player_events/starting_items.js:13]: Yeah, I think that's just github, but I'll try to make it a little less ugly here
image

[the remark this answers is not in the document]
Comment (MuteTiefling) [reply kubejs/server_scripts/expert/player_events/starting_items.js]: Should look better now. Tested and still working as intended.

File: kubejs/server_scripts/base/tags/fluids/minecraft/water.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ServerEvents.tags('fluid', (event) => {
+    event.get('minecraft:water').remove([/blood/, /tallow/, /sugar_water/]);
```
[the remark this answers is not in the document]
Comment (MuteTiefling) [reply kubejs/server_scripts/base/tags/fluids/minecraft/water.js:2]: Pretty sure this is a holdover from before forge fixed fluids. I can still swim fine in Blood, Tallow, and Sugar Water with this change, for example. 

Maybe we can leave it for Sugar Water though, it's close enough that tagging it as water makes sense.

[the remark this answers is not in the document]
Comment (NielsPilgaard) [reply kubejs/server_scripts/base/tags/fluids/minecraft/water.js:2]: Ah okay. Sounds good to me.

**Comment by NielsPilgaard** [comment]
I'd rather not, I spot them pre-release because they show up as git changes :P

**Comment by NielsPilgaard** [comment]
Awesome work, I like where this is going so far!

**Pull request merged**